### DISCUSSION TITLE
GitHub Action 在生成页面的过程中抛出No Layout导致所有页面变成0字节

### DISCUSSION THREAD

**Discussion by Eden7Ba23** [YunYouJun/hexo-theme-starry#103] [Q&A 疑难杂症] [answered]
[这是本人的Workflow](https://github.com/EdenJohnson2006/BlogBackup/blob/master/.github/workflows/deploy.yml)

经过~~差点虐死我~~的0基础Action编写之后，总算可以正常部署了

但是当我打开页面时，白花花一片 :(

本人采取的是部署过程中下载theme的方式，在查阅日志后发现下载theme后的generate过程中yun没有回报版本号，并且抛出了大量**No Layout**：

```sh
  /home/runner/.npm-global/bin/hexo g
  shell: /bin/bash -e {0}
INFO  Start processing
INFO  Files loaded in 286 ms
WARN  No layout: about/index.html
WARN  No layout: about/site.html
...
```

初步判断是yun没有正确下载导致的，但是在本地使用同样的指令后是能够形成正确的目录结构的：

```
eden@eden-arch  ~/test.yun     
git clone https://github.com/EdenJohnson2006/BlogBackup.git gitblog 
cd gitblog
git clone -b master https://github.com/YunYouJun/hexo-theme-yun themes/yun

正複製到 'gitblog'...
正複製到 'themes/yun'...
```

```
 eden@eden-arch  ~/test.yun/gitblog  master  tree -d   
├── source
│   ├── about
│   ├── categories
│   ├── _data
│   ├── links
│   ├── _posts
│   └── tags
└── themes
    └── yun
        ├── data
        ├── docs
        │   ├── en
        │   │   └── guide
        │   └── guide
        ├── languages
        ├── layout
        │   ├── _mixin
        │   ├── _partial
        │   │   ├── post
        │   │   └── sidebar
        │   ├── _third-party
        │   │   ├── adsense
        │   │   ├── analytics
        │   │   ├── comments
        │   │   ├── math
        │   │   └── seo
        │   └── _widget
        │       └── search
        ├── scripts
        │   ├── events
        │   ├── filters
        │   ├── helpers
        │   └── injector
        └── source
            ├── css
            │   ├── _components
            │   │   ├── post
            │   │   └── sidebar
            │   ├── _extend
            │   ├── _global
            │   ├── _layout
            │   ├── _mixins
            │   ├── _variables
            │   └── _widget
            │       └── search
            └── js
                ├── analytics
                ├── comments
                ├── search
                └── ui

50 directories
```

也是可以正常generate的

```
  eden@eden-arch  ~/test.yun/gitblog  master  hexo g
INFO  Yun version 1.4.0. Guide: https://yun.yunyoujun.cn
INFO  Start processing
INFO  Files loaded in 1.44 s
INFO  Generated: links.json
INFO  Generated: atom.xml
INFO  Generated: index.html
INFO  Generated: Rabbit-20210130.txt
...
```

想请教一下Author．

**Comment by YunYouJun** [comment, marked as answer]
```yaml
    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v2
```

actions/checkout 已经默认在本地 clone 了该项目，无需

```yaml
       run: |
          git clone https://github.com/EdenJohnson2006/BlogBackup.git gitblog
          cd gitblog
```

只需

```yaml
       run: |
          git clone -b master https://github.com/YunYouJun/hexo-theme-yun themes/yun
```

然后生成文件即可。

> https://github.com/EdenJohnson2006/BlogBackup/blob/90264fd01055da9bba33201a3956fe015705819f/.github/workflows/deploy.yml#L23



可以直接参考我的配置使用 `submodule` (去除 algolia)

> https://github.com/YunYouJun/yunyoujun.github.io/blob/hexo/.github/workflows/gh-pages.yml

**Reply by Eden7Ba23** [reply]
> ```yaml
>     steps:
>       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
>       - uses: actions/checkout@v2
> ```
> 
> actions/checkout 已经默认在本地 clone 了该项目，无需
> 
> ```yaml
>        run: |
>           git clone https://github.com/EdenJohnson2006/BlogBackup.git gitblog
>           cd gitblog
> ```
> 
> 只需
> 
> ```yaml
>        run: |
>           git clone -b master https://github.com/YunYouJun/hexo-theme-yun themes/yun
> ```
> 
> 然后生成文件即可。
> 
> > https://github.com/EdenJohnson2006/BlogBackup/blob/90264fd01055da9bba33201a3956fe015705819f/.github/workflows/deploy.yml#L23
> 
> 可以直接参考我的配置使用 `submodule` (去除 algolia)
> 
> > https://github.com/YunYouJun/yunyoujun.github.io/blob/hexo/.github/workflows/gh-pages.yml

感谢Author 在修改Workflow使其直接进入目录之后[问题已经得到解决](https://github.com/EdenJohnson2006/BlogBackup/runs/1814546783)，清除CloudFlare缓存之后网站恢复正常．
~~又是做了无用功的一天wwww~~

**Comment by zackzeng00** [comment]
使用其他主题遇到一样的问题.
搜索到你的答案,成功解决问题,感谢.

**Reply by bigcyy** [reply]
感谢，成功解决了

**Comment by YatingLo** [comment]
設定 submdule
https://github.com/YatingLo/YatingLo.github.io/blob/main/.gitmodules
actions/checkout@v2 需要設定 checkout submdules
https://github.com/YatingLo/YatingLo.github.io/blob/main/.github/workflows/pages.yml#L12-L14

**Reply by YunYouJun** [reply]
不再需要设定 submodule，请通过 npm 安装对应包的方式使用。

**Comment by wyxdlz54188** [comment]
你好我这边是从本地上传的也遇到了同样的问题，能不能远程帮忙搞一下
